python/pyiceberg/catalog/dynamodb.py

```diff
@@ -0,0 +1,776 @@
+    current_timestamp_ms = str(round(time() * 1000))
+
+    # noinspection PyDictCreation
+    _dict = {}
+    _dict[DYNAMODB_COL_IDENTIFIER] = namespace_item[DYNAMODB_COL_IDENTIFIER]
+    _dict[DYNAMODB_COL_NAMESPACE] = namespace_item[DYNAMODB_COL_NAMESPACE]
+    _dict[DYNAMODB_COL_VERSION] = {
+        "S": str(uuid.uuid4()),
+    }
+    _dict[DYNAMODB_COL_CREATED_AT] = namespace_item[DYNAMODB_COL_CREATED_AT]
+    _dict[DYNAMODB_COL_UPDATED_AT] = {
+        "N": current_timestamp_ms,
+    }
+
+    for key, val in updated_properties.items():
+        _dict[_add_property_prefix(key)] = {"S": val}
+
+    return _dict
+
+
+def _get_create_catalog_attribute_definitions() -> List[Dict[str, str]]:
+    return [
+        {
+            "AttributeName": DYNAMODB_COL_IDENTIFIER,
+            "AttributeType": "S",
+        },
+        {
+            "AttributeName": DYNAMODB_COL_NAMESPACE,
+            "AttributeType": "S",
+        },
+    ]
+
+
+def _get_key_schema() -> List[Dict[str, str]]:
+    return [
+        {
+            "AttributeName": DYNAMODB_COL_IDENTIFIER,
+            "KeyType": "HASH",
+        },
+        {
+            "AttributeName": DYNAMODB_COL_NAMESPACE,
+            "KeyType": "RANGE",
+        },
+    ]
+
+
+def _get_global_secondary_indexes() -> List[Dict[str, Any]]:
+    return [
+        {
+            "IndexName": DYNAMODB_NAMESPACE_GSI,
+            "KeySchema": [
+                {
+                    "AttributeName": DYNAMODB_COL_NAMESPACE,
+                    "KeyType": "HASH",
+                },
+                {
+                    "AttributeName": DYNAMODB_COL_IDENTIFIER,
+                    "KeyType": "RANGE",
+                },
+            ],
+            "Projection": {
+                "ProjectionType": "KEYS_ONLY",
+            },
+        }
+    ]
+
+
+def _get_namespace_properties(namespace_dict: Dict[str, str]) -> Properties:
+    namespace_properties = {}
+    for key, val in namespace_dict.items():
+        if not key.startswith(PROPERTY_KEY_PREFIX):
+            continue
+
+        namespace_properties[_remove_property_prefix(key)] = val
+    return namespace_properties
+
+
+def _convert_dynamo_item_to_regular_dict(dynamo_json: Dict[str, Any]) -> Dict[str, str]:
+    """
+    Converts a dynamo json to a regular json. Example of a dynamo json:
+    {
+        "AlbumTitle": {
+            "S": "Songs About Life",
+        },
+        "Artist": {
+            "S": "Acme Band",
+        },
+        "SongTitle": {
+            "S": "Happy Day",
+        }
+    }
+
+    Converted to regular json:
+    {
+        "AlbumTitle": "Songs About Life",
+        "Artist": "Acme Band",
+        "SongTitle": "Happy Day"
+    }
+
+    Only "S" and "N" data types are supported since those are the only ones that Iceberg is utilizing.
+    """
+
+    regular_json = {}
+    for column_name, val_dict in dynamo_json.items():
+        keys = list(val_dict.keys())
+        assert len(keys) == 1
```

Comment:
We tend to use assertions only in test code since they can be disabled. Let's throw a ValueError instead.

Reply:
Done.